includes/tracks/class-wc-tracks-client.php

```diff
@@ -92,10 +92,49 @@ public static function build_timestamp() {
 		return number_format( $ts, 0, '', '' );
 	}
 
+	/**
+	 * Get a user's identity to send to Tracks. If Jetpack exists, default to its implementation.
+	 *
+	 * @param int $user_id User id.
+	 * @return array Identity properties.
+	 */
+	public static function get_identity( $user_id ) {
+		if ( class_exists( 'Jetpack' ) ) {
+
+			include_once ABSPATH . 'wp-content/plugins/jetpack/_inc/lib/tracks/client.php';
+
+			if ( function_exists( 'jetpack_tracks_get_identity' ) ) {
+				return jetpack_tracks_get_identity( $user_id );
+			}
+		}
+
+		// Start with a previously set cookie.
+		$anon_id = isset( $_COOKIE['tk_ai'] ) ? wp_unslash( $_COOKIE['tk_ai'] ) : false; // phpcs:ignore WordPress.Security.ValidatedSanitizedInput.InputNotSanitized
+
+		// If there is no cookie, apply a saved id.
+		if ( ! $anon_id ) {
+			$anon_id = get_user_meta( $user_id, 'woo_tracks_anon_id', true );
+		}
+
+		// If an id is still not found, create one and save it.
+		if ( ! $anon_id ) {
+			$anon_id = self::get_anon_id();
+			update_user_meta( $user_id, 'woo_tracks_anon_id', $anon_id );
```

Comment:
We could rename it to `_woo_tracks_anon_id`, so will be hidden by default, also we probably need to exclude it from our CRUD too: https://github.com/woocommerce/woocommerce/blob/e91f3b88f58a7bcaa4e2a9f771621c7960d9a746/includes/data-stores/class-wc-customer-data-store.php#L25